jina/types/document/__init__.py

```diff
@@ -135,6 +137,12 @@ def __init__(self, document: Optional[DocumentSourceType] = None,
                              f'if you are trying to set the content '
                              f'you may use "Document(content=your_content)"') from ex
 
+        if custom_id is None:
+            import random
+            custom_id = random.randint(0, 9)
```

Comment:
So is the id a string (from the function signature) or an int?

Reply:
It does not matter. Good opportunity to dig a little deeper: Try to find out why is does not matter by looking at the implementation of `self.id`.